cylc/flow/subprocpool.py

```diff
@@ -300,6 +301,7 @@ def _poll_proc_pipes(self, proc, ctx):
                     data = os.read(fileno, 65536).decode()  # 64K
                 except OSError:
                     continue
+                received_data.append(data != '')
```

Comment:
So every few minutes an empty string is "pushed down the pipe"? (When it should be that nothing is pushed down?).

Reply:
It doesn't make any sense to me, the poller keeps producing the same file handles even though these are the stdout/err for a process which has long since completed. When we try to read from these files we get nothing (though strangely no OSError) hence the empty strings.

I think if the poller yields files which have no new data then the poller can be said to be broken so we can safely exit this loop. It would be nicer to fix the poller its self of course...